core/2d/CCParticleSystem.h

```diff
@@ -728,6 +780,136 @@ class CC_DLL ParticleSystem : public Node, public TextureProtocol, public Playab
+     * This function will automatically figure out your atlas cell size and direction for you! thank her later :) */
+    void setAnimationIndicesAtlas();
+
+    /** Add all particle animation indices based on cells size and direction spicified using a texture atlas.
+     * will erase the array and add new indices from the atlas.
+     *
+     * @param unifiedCellSize The size of cell unified.
+     * @param direction What direction is the atlas
+     */
+    void setAnimationIndicesAtlas(unsigned int unifiedCellSize, TexAnimDir direction = TexAnimDir::HORIZONTAL);
+
+    /** Add a particle animation index based on tex coords spicified using a sprite frame.
+     * The index is automatically incremented on each addition.
+     *
+     * @param frameName SpriteFrame name to search for
+     */
+    void addAnimationIndex(std::string_view frameName);
+
+    /** Add a particle animation index based on tex coords spicified using a sprite frame.
+     *
+     * @param frameName SpriteFrame name to search for
+     */
+    void addAnimationIndex(unsigned short index, std::string_view frameName);
+
+    /** Add a particle animation index based on tex coords spicified using a sprite frame.
+     * The index is automatically incremented on each addition.
+     * 
+     * @param frame SpriteFrame containting data about tex coords
+     */
+    void addAnimationIndex(cocos2d::SpriteFrame* frame);
+
+    /** Add a particle animation index based on tex coords spicified using a sprite frame.
+     * you can specify which index you want to override in this function
+     * @param index Index id to override the index with
+     * @param frame SpriteFrame containting data about tex coords
+     */
+    void addAnimationIndex(unsigned short index, cocos2d::SpriteFrame* frame);
+
+    /** Add a particle animation index based on tex coords spicified.
+     * you can specify which index you want to override in this function
+     * @param index Index id to override the index with
+     * @param rect Rect containting data about tex coords in pixels
+     * @param rotated Not implemented.
+     */
+    void addAnimationIndex(unsigned short index, cocos2d::Rect rect, bool rotated = false);
+
+    /** Add a particle animation descriptor with an index.
+    *
+    * @param indexOfDescriptor Index of the animation to be added, adding to the same index will just override the pervious animation descriptor
+    * @param time length of the animation in seconds
+    * @param timeVariance Time randomly selected for each different particle added on the animation length
+    * @param indices An array of the indicies
+    * @param reverse Should the animation indicies be played backwards? (default: false)
+    */
+    void setAnimationDescriptor(unsigned short indexOfDescriptor,
+                                float time,
+                                float timeVariance,
+                                std::vector<unsigned short> indices,
```

Comment:
std::vector use const reference maybe better

Reply:
You mean like `const std::vector<unsigned short> &indices`?

Reply:
Yep.